File: src/java_/problems/Problem0028.java
package java_.problems;

public class Problem0028 {

    public Problem0028() {
        long startTime = System.nanoTime();
        long result = 1;

        //schlau
        int step = 2;
        int num = 1;
        for (int i = 0; i < 500; i++) {
            for (int j = 0; j < 4; j++) {
                num += step;
                result += num;
            }
            step += 2;
        }

        long timeToResolve = System.nanoTime() - startTime;
        System.out.println("Result:\t" + result + "\tTime:\t" + (((double) timeToResolve / 1_000_000) > 1000 ?
                (((double) timeToResolve / 1_000_000_000) + "s") :
                (((double) timeToResolve / 1_000_000) + "ms")));
    }
}

File: src/main/cs/Problem0028.cs
﻿using System;
using System.Diagnostics;
using Euler.test.cs;

namespace Euler.main.cs
{
    public class Problem0028
    {
        public Problem0028()
        {
            Stopwatch stopWatch = new Stopwatch();
            stopWatch.Start();
            long result = 1;

            // Solution:
            int step = 2;
            int num = 1;
            for (int i = 0; i < 500; i++)
            {
                for (int j = 0; j < 4; j++)
                {
                    num += step;
                    result += num;
                }

                step += 2;
            }


            stopWatch.Stop();
            string elapsedTime = stopWatch.Elapsed.ToString();
            Console.WriteLine("Result:\t" + result + "\tTime:\t" +
                              (double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1, 2)) >= 1
                                  ? double.Parse(elapsedTime.Substring(elapsedTime.LastIndexOf(":") + 1)) + " s"
                                  : double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000 +
                                    " ms"));
            if (Test.DoBenchmark)
                Benchmark.AddTime(28, double.Parse(elapsedTime.Substring(elapsedTime.IndexOf(".") + 1)) / 10_000);
        }
    }
}